aptos-move/framework/supra-framework/sources/pbo_delegation_pool.move

```diff
@@ -865,7 +866,33 @@ module supra_framework::pbo_delegation_pool {
             enable_partial_governance_voting(pool_address);
         }
     }
+    
+    public entry fun fund_delegators_with_locked_stake(funder: &signer, pool_address: address, delegators: vector<address>, stakes: vector<u64>) acquires DelegationPool, BeneficiaryForOperator, GovernanceRecords, NextCommissionPercentage {
 
+
+        let  pool = borrow_global_mut<DelegationPool>(pool_address);
+        
+        vector::zip_reverse(delegators,stakes,|delegator,stake| {
```

Comment:
Why are you reversing? Shouldn't `zip` be sufficient?

Reply:
@isaacdoidge , I am not reversing anything. `zip_reverse` would access both vectors from the end, using `pop` which would be cheaper than forward pass using `borrow`. During `pop` it would consume/destroy the vector, which is fine here.

Reply:
Nice solution!